.github/workflows/build-test-and-publish.yaml

```diff
@@ -69,6 +69,10 @@ jobs:
           NEW_VERSION: ${{ steps.version.outputs.version }}
       - name: test and build JAR
         run: mvn package -ntp
+      - name: Upload new versions in maven registry
+        run: | 
+          cp settings.xml ~/.m2/
```

Comment:
Let's move settings.xml to .m2 folder in the repository to make it clean and use `--settings` option for `mvn` command to set the alternate path for the user settings file in the workflow

Reply:
Let's also use `-ntp` option to mute download progress.